src/bootstrap/src/utils/exec.rs

```diff
@@ -679,6 +754,8 @@ impl<'a> DeferredCommand<'a> {
                     && let (Some(cache_key), Some(_)) = (&cache_key, output.status())
                 {
                     exec_ctx.command_cache.insert(cache_key.clone(), output.clone());
+                    let duration = start_time.elapsed();
+                    exec_ctx.profiler.record(cache_key.clone(), duration);
```

Comment:
We should ideally record also the uncached commands. I guess that we could reuse the CacheKey logic to avoid introducing another approach for identifying the commands. So maybe let all commands return a cache key, and add a separate method for figuring out if caching is enabled or not? And maybe rename `CacheKey` to `CommandFingerprint` or something.